### DISCUSSION TITLE
help or feature req - "run_if" with logical "or"?

### DISCUSSION THREAD

**Discussion by miketwenty1** [bevyengine/bevy#8703] [Ideas]
If I have something like this:

```rs
//...
#[derive(Clone, Copy, Default, Eq, PartialEq, Debug, Hash, States)]
pub enum ApiState {
    #[default]
    Off,
    SetName,
    Move,
    // .. many more potential states
}

//...
app.init_resource::<ApiPollingTimer>()
    .add_systems(
        Update,
        tick_api_receive_timer.run_if(in_state(ApiState::SetName)),
```

And I know I'll want to run this `tick_api_receive_timer` for multiple `ApiState` states. It would be nice to be able to have something like one of these:

```rs
.run_if(in_state([ApiState::SetPlayerName, ApiState::MovePlayer])),
```
```rs
.run_if(in_state(ApiState::SetPlayerName).or(ApiState::MovePlayer)),
```
```rs
.run_if(in_state(ApiState::SetPlayerName)).or_if(in_state(ApiState::SetPlayerName)),
```

I'm unsure if "`in_set`" accomplishes this or some other thing I'm unfamiliar with. But I would potentially only want this to run the update if it's in a subset of the total possible ApiState. (i.e. not updating if the ApiState is set to "Off".)

**Comment by alice-i-cecile** [comment]
You can use the `or_else` run condition combinator for this. See https://github.com/bevyengine/bevy/issues/8702, which tracks adding an example of it to the docs :)

**Reply by miketwenty1** [reply]
aww.. snap! @alice-i-cecile 
i really thought i did my diligence on this one. I read the description of "and_then" but not "or_else" 
`Returns a new run condition that returns true if either this one or the passed or_else return true.`

For anyone curious it would look like this:
```rs
run_if(in_state(ApiState::SetName).or_else(in_state(ApiState::Move))
```
